src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -276,14 +276,11 @@ static string EncodeIndices(IEnumerable<int> indices, int additionalLocationsLen
             return $"https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/{id.ToLowerInvariant()}";
         }
 
-        public sealed class LocalizableStringWithArguments : LocalizableString, IObjectWritable
+        public sealed class LocalizableStringWithArguments : LocalizableString
```

Comment:
IObjectWritable as a concept goes away.  Instead of the serialization system taking in arbitrary objects, querying for that, and having that object then decide what to do, serialization is owned entirely on the outside by the objects, who just use our serialization system as a dumb stream wrapper to do things like write ints, bools, and strings to.  Only basic constructs are supported (like primitive types, and arrays of types).  Arbitrary object graphs are now handled by the caller (note: we have no such cases in roslyn anyways.  it was only 'syntax' with its heterogenous graph structure).

Reply:
LocalizableStirng used to have to be serializable, because it was stored in diagnostics as the messages, and diagnostics were stored inside a tree for syntax errors, and as such needed to be serializable if we serialized the tree.  Now that we no longer serialize trees, neither diagnostics nor locstrings need this functionality.